### PR TITLE
Implement pinvoke support for versioned symbols

### PR DESCRIPTION
**Pull request by tmds**
Implement #8721 

This is an initial implementation which uses a 'symbol@version' EntryPoint API and includes a test.

I manually compiled and ran the test on my linux system (because of the Windows build dependency).

I am sending the PR to get feedback and to see how behaves on the different CI slaves (especially non-Linux platforms).

CC @jkotas @yizhang82

### CODE REVIEW

File: tests/src/Interop/CMakeLists.txt [hunk 1]
```diff
@@ -26,3 +26,4 @@ add_subdirectory(StringMarshalling/UTF8)
 add_subdirectory(MarshalAPI/FunctionPointer)
 add_subdirectory(MarshalAPI/IUnknown)
 add_subdirectory(SizeConst)
+add_subdirectory(SymVer)
```
Comment (tmds) [inline tests/src/Interop/CMakeLists.txt]: Excluding the native project on Windows gives a build error:
```
08:10:21 D:\j\workspace\checked_windo---f6dc6fe4\tests\src\dir.targets(126,6): error : The native project files are missing in D:\j\workspace\checked_windo---f6dc6fe4\bin\int\Native\src\Interop\SymVer\Checked\ please run build from the root of the repo at least once [D:\j\workspace\checked_windo---f6dc6fe4\tests\src\Interop\SymVer\SymVer.csproj]
```

File: tests/src/Interop/SymVer/SymVerManaged.cs [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+            try
+            {
+                foo_11();
+            }
+            catch (Exception)
+            {
+                exceptionThrown = true;
+            }
+            Assert(exceptionThrown, "Call to foo_11 should throw an exception");
+            // TODO: check for more specific exception
```
Comment (tmds) [inline tests/src/Interop/SymVer/SymVerManaged.cs]: Probably this is throwing EntryPointNotFoundException now, but PlatformNotSupportedException may be a better fit.

**Comment by tmds** [comment]
I don't know why the OSX build failed. Can we trigger it again?

**Comment by janvorli** [comment]
@dotnet-bot test OSX x64 Checked Build and Test please

**Comment by tmds** [comment]
@dotnet-bot test Ubuntu x64 Checked Build and Test please

**Comment by danmosemsft** [comment]
@tmds is this ready for final code review?

**Comment by jkotas** [comment]
@danmosemsft This is being discussed in https://github.com/dotnet/coreclr/issues/8721

**Comment by karelz** [comment]
Looks like we still don't have consensus. Closing the PR for now, we can reopen it when the change is approved / ready for further work.

**Pull request closed without merging**